### ISSUE TITLE
Logs for callback are empty 

### ISSUE THREAD

**Issue by ghost** [keras-team/keras#10498]
Hi everyone,

I have an issue during the callback process. Maybe one of you could help me out ?
I'm trying to make my own callback. Here is the code :

class EarlyStoppingByLossVal(keras.callbacks.Callback):

           def init(self, monitor='val_loss', close_enough=1,too_far_away=50, verbose=0):
                      super(keras.callbacks.Callback, self).init()
                      self.monitor = monitor
                      self.close_enough = close_enough
                      self.too_far_away = too_far_away
                      self.verbose = verbose

           def on_epoch_begin(self, epoch, logs={}):
                     current =  logs.get(self.monitor)

                      if current is not None:
                              if current > self.too_far_away:
                                     print("Epoch %05d: Starting too far" % epoch)
                                      self.model.stop_training = True

           def on_epoch_end(self, epoch, logs={}):
                     current = logs.get(self.monitor)

                      if current is not None:
                              if current < self.close_enough:
                                       print("Epoch %05d: early stopping THR" % epoch)
                                      self.model.stop_training = True

           model = Sequential()
           model.add(Dense(units=10, activation='relu', input_shape=(3,)))
           model.add(Dense(units=10, activation='relu', ))
           model.add(Dense(units=10, activation='relu', ))
           model.add(Dense(units=10, activation='relu', ))
           model.add(Dense(units=10, activation='relu', ))
           model.add(Dense(units=10, activation='relu', ))
           model.add(Dense(units=10, activation='relu', ))
           model.add(Dense(units=10, activation='relu', ))
           model.add(Dense(units=10, activation='relu', ))
           model.add(Dense(units=1, activation='relu'))

           model.compile(loss='mse', optimizer='adam',metrics=['mae'])

           callbacks = [
           EarlyStoppingByLossVal(monitor='val_loss', close_enough=1.5,too_far_away=50, verbose=1)
           ]

           model.fit(X_Train, Y_Train, epochs=5000, batch_size=4,
           validation_data=(X_Test, Y_Test),
           callbacks=callbacks)

No matter is in the label "monitor", my logs are always empty.

Do you have any explication for this ?

Thanks a lot !
